app/src/main/java/com/example/teamyuml/decentworkmobile/views/EngagmentAdd.java

```diff
@@ -0,0 +1,146 @@
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_advertisement_form);
+        advertisment_title_input = findViewById(R.id.Advertisment_Title);
+        advertisment_description_input = findViewById(R.id.Advertiser_Description);
+        advertisment_city = findViewById(R.id.Advertiser_City);
+        ErrorTextView = findViewById(R.id.textView5);
+        Cities = new ArrayList<String>();
+        Cities.add("Wybierz miasto");
+        Cities.add("Łódź");
+        Cities.add("Poznań");
+        Cities.add("Wrocław");
+        ArrayAdapter<String> arrayAdapter = new ArrayAdapter<String>(EngagmentAdd.this,android.R.layout.simple_dropdown_item_1line,Cities);
+        arrayAdapter.setDropDownViewResource(android.R.layout.simple_dropdown_item_1line);
+        advertisment_city.setAdapter(arrayAdapter);
+        advertisment_city.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
+                selected_city = advertisment_city.getItemAtPosition(i).toString();
+            }
+            @Override
+            public void onNothingSelected(AdapterView<?> adapterView) {
+            }
+        });
+        sign_Advertisment = findViewById(R.id.Advertiser_Add);
+        sign_Advertisment.setOnClickListener(this);
+    }
+
+    @Override
+
+    /**
+     * Makes request to server and add engagment if success or shows
+     * error when failure.
+     */
+    public void onClick(View v)
+    {
+        JSONObject data = this.addParams();
+        JsonObjectRequest jsonObjectRequest = new JsonObjectRequest
+        (Request.Method.POST, Engagment_add_URL, data, new Response.Listener<JSONObject>() {
+
+            @Override
+            public void onResponse(JSONObject response) {
+
+                try {
+                    String owner = response.getString("owner");
+                    ErrorTextView.setText("pomyslnie dodano" + owner);
+                }
+                catch (JSONException e) {
+                    e.printStackTrace();
+                }
+            }
+        },  new Response.ErrorListener() {
+
+            @Override
+              public void onErrorResponse(VolleyError error) {
+                ErrorTextView.setText(error.getMessage());
+                Log.e(TAG, "Adding Error: " + error.getMessage());
+              }
+        })
+        {
+            public Map<String, String> getHeaders() throws AuthFailureError {
+
+                return UserAuth.authorizationHeader(EngagmentAdd.this);
+            }
+        };
+        // Adding request to request queue
+        VolleyInstance.getInstance(this).addToRequestQueue(jsonObjectRequest, "Create Engagment");
+    }
+
+    /**
+     * Method is responsibility for adding params to engagment
+     * @return CreateJson object
+     */
+    public JSONObject addParams() {
```

Comment:
I think this should be private, cause we don't want to use it in any other class